fdbserver/LocalConfiguration.actor.cpp

```diff
@@ -0,0 +1,395 @@
+			wait(saveConfigPath(self));
+			self->updateInMemoryState(lastSeenVersion);
+			return Void();
+		}
+		state ConfigKnobOverrides storedConfigPath =
+		    BinaryReader::fromStringRef<ConfigKnobOverrides>(storedConfigPathValue.get(), IncludeVersion());
+		if (!storedConfigPath.hasSameConfigPath(self->configKnobOverrides)) {
+			// All local information is outdated
+			wait(clearKVStore(self));
+			wait(saveConfigPath(self));
+			self->updateInMemoryState(lastSeenVersion);
+			return Void();
+		}
+		Standalone<RangeResultRef> range = wait(self->kvStore->readRange(knobOverrideKeys));
+		for (const auto& kv : range) {
+			auto configKey =
+			    BinaryReader::fromStringRef<ConfigKey>(kv.key.removePrefix(knobOverrideKeys.begin), IncludeVersion());
+			self->configKnobOverrides.set(configKey.configClass,
+			                              configKey.knobName,
+			                              ObjectReader::fromStringRef<KnobValue>(kv.value, IncludeVersion()));
+		}
+		self->updateInMemoryState(lastSeenVersion);
+		return Void();
+	}
+
+	void updateInMemoryState(Version lastSeenVersion) {
+		this->lastSeenVersion = lastSeenVersion;
+		// TODO: Support randomization?
+		getKnobs().reset(Randomize::NO, g_network->isSimulated() ? IsSimulated::YES : IsSimulated::NO);
+		configKnobOverrides.update(getKnobs());
+		manualKnobOverrides.update(getKnobs());
+		// Must reinitialize in order to update dependent knobs
+		getKnobs().initialize(Randomize::NO, g_network->isSimulated() ? IsSimulated::YES : IsSimulated::NO);
+	}
+
+	ACTOR static Future<Void> setSnapshot(LocalConfigurationImpl* self,
+	                                      std::map<ConfigKey, KnobValue> snapshot,
+	                                      Version snapshotVersion) {
+		// TODO: Concurrency control?
+		ASSERT(self->initFuture.isValid() && self->initFuture.isReady());
+		++self->snapshots;
+		self->kvStore->clear(knobOverrideKeys);
+		for (const auto& [configKey, knobValue] : snapshot) {
+			self->configKnobOverrides.set(configKey.configClass, configKey.knobName, knobValue);
+			self->kvStore->set(
+			    KeyValueRef(BinaryWriter::toValue(configKey, IncludeVersion()).withPrefix(knobOverrideKeys.begin),
+			                ObjectWriter::toValue(knobValue, IncludeVersion())));
+		}
+		ASSERT_GE(snapshotVersion, self->lastSeenVersion);
+		self->kvStore->set(KeyValueRef(lastSeenVersionKey, BinaryWriter::toValue(snapshotVersion, IncludeVersion())));
+		wait(self->kvStore->commit());
+		self->updateInMemoryState(snapshotVersion);
+		return Void();
+	}
+
+	ACTOR static Future<Void> addChanges(LocalConfigurationImpl* self,
+	                                     Standalone<VectorRef<VersionedConfigMutationRef>> changes,
+	                                     Version mostRecentVersion) {
+		// TODO: Concurrency control?
+		ASSERT(self->initFuture.isValid() && self->initFuture.isReady());
+		++self->changeRequestsFetched;
+		for (const auto& versionedMutation : changes) {
```

Comment:
Add an assertion that `versionedMutation.version` is greater than or equal to `self->lastSeenVersion)`.